client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx

```diff
@@ -60,6 +61,15 @@ export function DomainCodePair( {
 
 	const shouldReportError = hasDuplicates || ( ! loading && domain && auth ? true : false );
 
+	useEffect( () => {
+		if ( shouldReportError ) {
+			recordTracksEvent( 'calypso_domain_transfer_domain_error', {
+				domain,
+				message,
+			} );
+		}
```

Comment:
`shouldReportError` doesn't really imply there is an error, it means if there is one, we'd show it to the user. I added it to hide validation errors when domains are empty. 
```suggestion
		if ( shouldReportError && ! valid && message ) {
			recordTracksEvent( 'calypso_domain_transfer_domain_error', {
				domain,
				message,
			} );
		}
```